src/components/Subscribe/Subscribe.tsx

```diff
@@ -0,0 +1,286 @@
+    const [subscribeSelectors, setSubscribeSelectors] = useState(SubscribeType.documentation);
+
+    const hideSubscribePopups = useCallback(() => {
+        setShowSubscribeSuccessPopup(false);
+        setShowSubscribeVariantsPopup(false);
+        setShowError(false);
+        setMail('');
+    }, []);
+
+    const resetSubscribeAdditionalInfo = useCallback(() => {
+        setMail('');
+    }, []);
+
+    useEffect(() => {
+        i18n.changeLanguage(lang);
+    }, [i18n, lang]);
+
+    const setTimer = useCallback((callback: () => void) => {
+        timerId.current = setTimeout(async () => {
+            callback();
+        }, timeout);
+    }, []);
+
+    const clearTimer = useCallback(() => {
+        clearTimeout(timerId.current as number);
+        timerId.current = undefined;
+    }, []);
+
+    useEffect(() => {
+        if (showSubscribeSuccessPopup) {
+            setTimer(() => {
+                setShowSubscribeSuccessPopup(false);
+                clearTimer();
+            });
+        }
+    }, [clearTimer, setTimer, showSubscribeSuccessPopup]);
+
+    const setSubscribeControlRef = useCallback((ref) => {
+        subscribeControlRef.current = ref;
+    }, []);
+
+    const onOutsideClick = useCallback(() => {
+        hideSubscribePopups();
+        setShowError(false);
+    }, [hideSubscribePopups]);
+
+    const onSendSubscribeInformation = useCallback(() => {
+        if (isEmailHasAtSymbol(mail)) {
+            setShowSubscribeSuccessPopup(true);
+            setShowSubscribeVariantsPopup(false);
+            setShowError(false);
+
+            if (onSubscribe) {
+                onSubscribe({
+                    mail,
+                    type: subscribeSelectors,
+                });
+
+                resetSubscribeAdditionalInfo();
+            }
+        } else {
+            setShowError(true);
+        }
+    }, [onSubscribe, mail, resetSubscribeAdditionalInfo, subscribeSelectors]);
+
+    const getPopupPosition = useCallback(() => {
+        if (!view || view === SubscribeView.Regular) {
+            return isVerticalView ? PopperPosition.LEFT_START : PopperPosition.BOTTOM_END;
+        }
+
+        return PopperPosition.RIGHT;
+    }, [isVerticalView, view]);
+
+    const onChangeSubscribe = () => {
+        setShowSubscribeVariantsPopup(true);
+        setShowSubscribeSuccessPopup(false);
+    };
+
+    const renderSubscribeControl = useCallback(() => {
+        return (
+            <Control
+                onClick={onChangeSubscribe}
+                className={b('control', {view}, classNameControl)}
+                isVerticalView={isVerticalView}
+                tooltipText={t(`subscribe-text`)}
+                setRef={setSubscribeControlRef}
+                icon={(args) => <SubscribeIcon {...args} />}
+            />
+        );
+    }, [classNameControl, view, isVerticalView, setSubscribeControlRef, t]);
+
+    const renderRegularSubscribeControls = useCallback(() => {
```

Comment:
Why this cb is needed?
It only wraps `renderSubscribeControl` and unexpectedly uses Fragment.